futures-util/src/future/mod.rs

```diff
@@ -554,30 +591,30 @@ pub trait FutureExt: Future {
     /// # Examples
     ///
     /// ```
-    /// # extern crate futures;
-    /// use futures::prelude::*;
-    /// use futures::future;
-    /// use futures::executor::block_on;
+    /// #![feature(async_await, await_macro, futures_api)]
+    /// # futures::executor::block_on(async {
+    /// use futures::future::{self, FutureExt};
     ///
-    /// # fn main() {
     /// let future = future::ready(6);
     /// let shared1 = future.shared();
     /// let shared2 = shared1.clone();
     ///
-    /// assert_eq!(6, *block_on(shared1));
-    /// assert_eq!(6, *block_on(shared2));
-    /// # }
+    /// assert_eq!(6, *await!(shared1));
+    /// assert_eq!(6, *await!(shared2));
+    /// # });
     /// ```
     ///
     /// ```
-    /// # extern crate futures;
+    /// // Note, unlike most examples this is written in the context of a
+    /// // synchronous function to better illustrate the cross-thread aspect of
+    /// // the `shared` combinator.
+    ///
     /// use std::thread;
```

Comment:
Do we really want separate std imports? During my refactorings for 0.3 I made it so that all imports are simply always bundled together alphabetically ordered and `std` is treated like any other crate.

Reply:
Sure, I don't really mind either way.